extensions/dub-link-shortener/src/api/index.ts

```diff
@@ -1,66 +1,47 @@
-import axios, { AxiosRequestConfig } from "axios";
-import { BASE_API_URL } from "@utils/constants";
-import { apiKey, commandName, extensionName } from "@utils/env";
-import { DeleteLinkResponseBody, LinkSchema, TagSchema, WorkspaceId, WorkspaceSchema } from "@/types";
-import { captureException } from "@raycast/api";
+import { Dub } from "dub";
+import { apiKey } from "@utils/env";
 
-const hasHttps = (url: string) => url.startsWith("https://");
-const headers = {
-  Authorization: "Bearer " + apiKey,
-  "Content-Type": "application/json",
-  "user-agent": `raycast/${extensionName}/${commandName}`,
-};
-
-/**
- * todo: Replace with SDK https://d.to/sdk once it is stable
- * see: https://github.com/dubinc/dub-node/blob/765e170c45a361de3ae62e0d19571ceca4a3f0f4/README.md#maturity
- */
-export const callApi = async <T>(config: AxiosRequestConfig) => {
-  return await axios({ ...config, headers })
-    .then(({ data }) => data as T)
-    .catch(({ response }) => {
-      const err = new Error(response?.data?.error?.message ?? "Unknown error");
-      captureException(err);
-      throw err;
-    });
-};
-
-export const getAllWorkspaces = async () => {
-  return await callApi<WorkspaceSchema[]>({ method: "GET", url: `${BASE_API_URL}/workspaces` });
-};
+const dub = new Dub({
+  token: apiKey,
+});
 
 export const createShortLink = async ({
   originalUrl,
   key,
-  workspaceId,
   domain,
   tagIds,
   comments,
-}: { originalUrl: string; key?: string; domain?: string; tagIds?: string[]; comments?: string } & WorkspaceId) => {
-  const url = hasHttps(originalUrl) ? originalUrl : `https://${originalUrl}`;
-  return await callApi<LinkSchema>({
-    method: "POST",
-    url: workspaceId ? `${BASE_API_URL}/links?workspaceId=${workspaceId}` : `${BASE_API_URL}/links`,
-    data: { domain, url, key, tagIds, comments },
+}: {
+  originalUrl: string;
+  key?: string;
+  domain?: string;
+  tagIds?: string[];
+  comments?: string;
+}) => {
+  return dub.links.create({
+    url: originalUrl,
```

Comment:
Question (non-blocking): Was the `hasHttps` check not required?

Reply:
yup, we handle this on the backend automatically via Zod + the URL API: https://github.com/dubinc/dub/blob/main/apps/web/lib/zod/schemas/utils.ts#L4-L7